### DISCUSSION TITLE
Device Identification

### DISCUSSION THREAD

**Discussion by praveshgupta1586** [talsec/Free-RASP-Community#35] [General]
Dear Team,
We integrated "freerasp" in our application and we are getting reports also. However we are not able to identify the device id from our system with shared report by "freerasp". Our application generates a seprate device ID using vendorID in iOS and Settings.Secure.ANDROID_ID in android. But the " freerasp" reports show different device IDs so we are unable to identify the effected devices.
Please help us to identify the device from report.

**Comment by msikyna** [comment]
Hello @praveshgupta1586 ,
the identifiers are:
- iOS: identifierForVendor
- Android: Android ID

Therefore, you should be able to match those. However, if you want to also provision some additional identifiers, the Business RASP+ offers this option, by setting your own 'externalID'. For more information about BusinessRASP+, see the [Differences between freeRASP vs Business RASP+](https://github.com/orgs/talsec/discussions/5).

Kind regards,
Talsec team
